functional-tests/src/test/java/org/infinispan/online/service/caching/AbstractBasicTest.java

```diff
@@ -1,58 +1,44 @@
 package org.infinispan.online.service.caching;
 
 import io.fabric8.openshift.client.OpenShiftClient;
-import org.arquillian.cube.kubernetes.annotations.Named;
-import org.arquillian.cube.openshift.impl.requirement.RequiresOpenshift;
-import org.arquillian.cube.requirement.ArquillianConditionalRunner;
 import org.infinispan.online.service.endpoint.HotRodTester;
 import org.infinispan.online.service.endpoint.RESTTester;
+import org.infinispan.online.service.utils.OpenShiftHandle;
 import org.infinispan.online.service.utils.ReadinessCheck;
-import org.jboss.arquillian.test.api.ArquillianResource;
-import org.junit.Before;
 import org.junit.Ignore;
 import org.junit.Test;
-import org.junit.runner.RunWith;
 
 import java.io.IOException;
 import java.net.URL;
+import java.util.Random;
 import java.util.concurrent.TimeUnit;
 
-@RunWith(ArquillianConditionalRunner.class)
-@RequiresOpenshift
-public class CachingServiceTest {
+public abstract class AbstractBasicTest {
```

Comment:
I've been talking about test inheritance with @ryanemerson and I would like to at least try to avoid test inheritance in this project. 

What I often see in Infinispan is that we have layers upon layers of inherited classes in tests. This may seem like a good idea at first but then it evolves pretty ugly. Later on we extract some bits of the base class class turn it off and on in specialized test cases. In addition to all of that, lots of base classes contain very good utility methods and that makes it almost impossible to write a test class without extending anything.

I would really love to try to do it differently here - create a set of smaller utilities and custom assertions. I think there is no shame in duplicating some small and lean tests (if both Caching and Shared Memory Service behave the same way in some aspects) rather than creating an abstract test class. In longer terms those short (even duplicated) tests are much easier to maintain in my opinion. 

WDYT?

//cc @ryanemerson @rmacior

Reply:
Makes sense. The inheritance also has unpredictable behavior when using together with Arquillian so I'd be happy to avoid that.